### ISSUE TITLE
Problems of the column filter functionality in v1.5.1 

### ISSUE THREAD

**Issue by liqili** [AllenFang/react-bootstrap-table#283]
Followed by example,I set the TableHeaderColumn  parameters like below.
`<TableHeaderColumn dataField="templateName" dataSort={true} filter={{type: "TextFilter", placeholder: "Please enter a value"}}>Report Name</TableHeaderColumn>`
But,I got errors when entering some text in the filter component.Errors illustrated as below.
"Uncaught TypeError: Cannot read property 'handleFilter' of undefined"
caused by: this.props.filter.emitter.handleFilter(this.props.dataField, value, type);

**Comment by AllenFang** [comment]
@liqili. could you try to remove the ```dataSort```? and post result  here. Thanks

**Comment by liqili** [comment]
@AllenFang I still got errors although removing the `dataSort`.Just like this.
`<BootstrapTable 
	      		id = "selectTable"
	      		data={this.dataSet} 
	      		keyField="id" 
				exportCSV={true}
				csvFileName = "template_sheet.csv"
	      		striped={true}
				hover={true}
				// condensed={true}
				pagination={true}
				search={true}
				selectRow={this.state.selectRowProp}>
			  <TableHeaderColumn dataField="id" hidden={true}>No.</TableHeaderColumn>
			  <TableHeaderColumn dataField="templateName" filter={{type: "TextFilter", placeholder: "Please enter a value"}}>Report Name</TableHeaderColumn>
			</BootstrapTable>`

**Comment by AllenFang** [comment]
ok, I'll check it out, thanks your feedback!

**Comment by dana2208** [comment]
@liqili, can you try not to hide the first column and see if it works?

**Comment by AllenFang** [comment]
I know the problem, is a bug you assign key column by ```keyField```, it's a bug currently, I will fix in next version. The workaround is add ```isKey={true}``` on ```<TableHeaderColumn>``` instead of ```keyField``` on ```<BootstrapTable>```. Like following:
```js
<BootstrapTable 
	      		id = "selectTable"
	      		data={this.dataSet} 
		        exportCSV={true}
			csvFileName = "template_sheet.csv"
	      		striped={true}
				hover={true}
				// condensed={true}
				pagination={true}
				search={true}
				selectRow={this.state.selectRowProp}>
			  <TableHeaderColumn dataField="id" isKey={true} hidden={true}>No.</TableHeaderColumn>
			  <TableHeaderColumn dataField="templateName" filter={{type: "TextFilter", placeholder: "Please enter a value"}}>Report Name</TableHeaderColumn>
			</BootstrapTable>
```

**Comment by AllenFang** [comment]
Fixed on v1.5.2
